### DISCUSSION TITLE
Enum error: "" is not a valid backing value for enum

### DISCUSSION THREAD

**Discussion by samhaiden** [filamentphp/filament#12125] [Help] [answered]
### Package

Panel builder

### Package Version

3.2.5

### How can we help you?

I am using enum classes for a variety of form options related to a users profile.

In a panel resource builder, if a user has previously saved a value and then selects 'Select an option' in the form dropdown menu, no errors are thrown. However, on a custom page, when a user takes this same action using the same component this error occurs: "" is not a valid backing value for enum

Example of Select component:

`Select::make('user.contact_time')
     ->label('Preferred Contact Time')
     ->options(ContactTimeEnums::class),`

I have attempted using various lifecycle hook methods to set the value to null but nothing seems to work. Any help would be appreciated.

**Comment by VincvanAs** [comment, marked as answer]
Laravel is expecting either a valid enum value or a null, and its getting neither.

Here's a quick fix: just sneak in a mutator in your model. It's like catching the empty string before it causes trouble and swapping it with null. For example put this into your model:

```
public function setContactTimeAttribute($value)
{
    $this->attributes['contact_time'] = $value === '' ? null : $value;
}
```

**Reply by samhaiden** [reply]
Thank you! I still don't understand why I wouldn't encounter the error when updating a filament resource page compared to the custom page but, regardless, this solution works 👍🏻
